tests/rbd/test_rbd_migration_external_nbd.py

```diff
@@ -0,0 +1,334 @@
+            for image_name, image_conf in multi_image_config.items():
+                try:
+                    # Create a target pool where image needs to be migrated on cluster2
+                    is_ec_pool = True if "ec" in pool_type else False
+                    config = kw.get("config", {})
+                    target_pool = "target_pool_" + random_string(len=5)
+                    target_pool_config = {}
+                    pools_to_delete = [target_pool]
+                    if is_ec_pool:
+                        data_pool_target = "data_pool_new_" + random_string(len=5)
+                        target_pool_config["data_pool"] = data_pool_target
+                        pools_to_delete.append(data_pool_target)
+
+                    io_rc = run_io_and_check_rbd_status(
+                        rbd=rbd,
+                        pool=pool,
+                        image=image_name,
+                        client=c1_client,
+                        image_conf=image_conf,
+                    )
+                    if io_rc:
+                        log.error(f"IO on image {image_name} failed")
+                        return 1
+
+                    md5_sum_before_migration = get_md5sum_rbd_image(
+                        image_spec=f"{pool}/{image_name}",
+                        rbd=rbd,
+                        client=c1_client,
+                        file_path=f"/tmp/{random_string(len=3)}",
+                    )
+                    log.info(f"md5sum before Migration: {md5_sum_before_migration}")
+
+                    unix_socket_path = "/tmp/nbd_socket_" + random_string(len=3)
+
+                    out, err = rbd.map(**{"image-or-snap-spec": f"{pool}/{image_name}"})
+                    device = out.strip()
+
+                    out, err = c1_client.exec_command(
+                        sudo=True,
+                        cmd=f"qemu-nbd -f raw --shared 10 --persistent --fork --socket {unix_socket_path} {device}",
+                    )
+                    if err:
+                        raise Exception(
+                            "Failed to export the RBD device as NBD server over unix socket"
+                        )
+                        return 1
+                    else:
+                        log.info(
+                            "Successfully exported the RBD device as NBD server over unix socket"
+                        )
+
+                    out, err = c1_client.exec_command(
+                        sudo=True,
+                        cmd=f"chmod 666 {unix_socket_path}",
+                    )
+                    if err:
+                        raise Exception(
+                            "Failed to change the permissions of unix socket"
+                        )
+                        return 1
+                    else:
+                        log.info("Successfully changed the permissions of unix socket")
+
+                    kw["cleanup_files"].append(unix_socket_path)
```

Comment:
`cleanup_files` are not being removed. Please ensure that all files created under this module are included, and handle their removal in the finally block so they get cleaned up even in case of a failure.